pkg/security/probe/probe.go

```diff
@@ -978,6 +978,78 @@ func (p *Probe) SelectProbes(rs *rules.RuleSet) error {
 	return p.manager.UpdateActivatedProbes(activatedProbes)
 }
 
+// DumpDiscarders removes all the discarders
+func (p *Probe) DumpDiscarders() (string, error) {
+	log.Debug("Dumping discarders")
+
+	dump, err := os.CreateTemp("/tmp", "discarder-dump-")
+	if err != nil {
+		return "", err
+	}
+	defer dump.Close()
+
+	if err := os.Chmod(dump.Name(), 0400); err != nil {
+		return "", err
+	}
+
+	fmt.Fprintf(dump, "Discarder Dump\n%s\n", time.Now().UTC().String())
+	fmt.Fprintf(dump, "\nInode Discarders\n")
+
+	discardedInodeCount := 0
+	var inode inodeDiscarderMapEntry
+	var inodeParams inodeDiscarderParams
+	for entries := p.inodeDiscarders.Iterate(); entries.Next(&inode, &inodeParams); {
```

Comment:
there is a risk here of having the map modified during the iteration which can break the iteration. In order to ensure that we won't fall into an infinite loop I would check that the `discarderInodeCount` is always lower than the map size. what do you think ?

Reply:
There is no easy way of counting the actual number of elements in the map, do you mean the "max_entries" ?